modules/components/src/AdvancedFacetView/Facets/TermAggregation.js

```diff
@@ -2,24 +2,41 @@ import React from 'react';
 import TermAggs from '../../Aggs/TermAgg';
 import { inCurrentSQON, replaceSQON, toggleSQON } from '../../SQONView/utils';
 
-export default ({ aggType, aggProps, title, onValueChange, sqon, path }) => (
-  <TermAggs
-    key={title}
-    collapsible={false}
-    handleValueClick={({ bucket }) => {
-      onValueChange({
-        value: bucket.key,
-      });
-    }}
-    field={path}
-    buckets={aggProps ? aggProps.buckets || [] : []}
-    displayName={title}
-    isActive={d => {
-      return inCurrentSQON({
-        value: d.value,
-        dotField: d.field,
-        currentSQON: sqon,
-      });
-    }}
-  />
-);
+export default ({
+  aggType,
+  aggProps,
+  title,
+  onValueChange,
+  sqon,
+  path,
+  maxTerms = 6,
+  stackingLengthLimit = 50,
+}) => {
+  const buckets = aggProps ? aggProps.buckets || [] : [];
+  const hasLongValues =
+    buckets.find(({ key }) => key.length > stackingLengthLimit) !== undefined;
```

Comment:
you don't need the undefined check

```
[1, 2, 3].find(x => x === 5)
// undefined
```

Reply:
I should print out a list of array HOF :P

Reply:
here you go https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array